### DISCUSSION TITLE
Create Snapshot with CommandInterpreter pattern?

### DISCUSSION THREAD

**Discussion by datasatanic** [dotnet/dotNext#66] [Q&A] [answered]
In  [docs](https://sakno.github.io/dotNext/features/cluster/wal.html#command-interpreter) not enough information about Snapshot. How I understand snapshot is simple command (implementing serialization/desirialization in CommandFormatter etc). How concatinate WriteToAsync in inherit from SnapshotBuilder and Command?

**Comment by sakno** [comment, marked as answer]
Hi @datasatanic , you need to use `CommandHandlerAttribute` with `IsSnapshotHandler` as mentioned in the documentation. This allows to capture snapshot entry when replaying log at application startup. `SnapshotBuilder` is ephemeral log entry so you need it only for implementing snapshot construction logic. Inside of snapshot builder you can construct a command representing the snapshot and call your formatter from `WriteToAsync` implementation.
